File: pc前端代码/melhat_pc-main/src/api/fence.js
import request from "@/utils/request";

// 分页查询电子围栏
export function getFencePage(query) {
  return request({
    url: "/hat/electronic/fence/page",
    method: "get",
    params: query,
  });
}

// 新增电子围栏
export function addFence(data) {
  return request({
    url: "/hat/electronic/fence",
    method: "post",
    data: data,
  });
}

// 修改电子围栏
export function updateFence(data) {
  return request({
    url: "/hat/electronic/fence",
    method: "put",
    data: data,
  });
}

// 查询电子围栏详情
export function getFence(id) {
  return request({
    url: "/hat/electronic/fence/" + id,
    method: "get",
  });
}

// 删除电子围栏
export function deleteFence(id) {
  return request({
    url: "/hat/electronic/fence/" + id,
    method: "delete",
  });
}

// 报警列表查询
export function getFenceAlarmPage(query) {
  return request({
    url: "/hat/fence/alarm/page",
    method: "get",
    params: query,
  });
}

File: 新版安卓端开发/网页展示/展示源码/lib/api/fence.dart
import 'package:rolling_intelligence_headband/http/index.dart';
import 'package:rolling_intelligence_headband/http/request_options.dart';
import 'package:rolling_intelligence_headband/http/response/page.dart';
import 'package:rolling_intelligence_headband/models/fence.dart';

class FenceApi {
  /// 电子围栏分页列表
  static Future<DataPage<Fence>> getFencePage({
    int? current,
    int? size,
    String? fenceName,
    String? fenceType,
    String? startTime,
    String? endTime,
  }) async {
    final response = await http.request(
      ReqOptions(
        path: '/hat/electronic/fence/page',
        method: 'GET',
        params: {
          'current': current,
          'size': size,
          'fenceName': fenceName,
          'fenceType': fenceType,
          'startTime': startTime,
          'endTime': endTime,
        },
      ),
    );
    return DataPage.fromJson(response, Fence.fromJson);
  }

  /// 保存电子围栏
  static Future<dynamic> saveFence(Fence fence) async {
    final response = await http.request(
      ReqOptions(
        path: '/hat/electronic/fence',
        method: 'POST',
        data: {
          'fence': fence.toJson(),
          'coordinates': fence.coordinates?.map((e) => e.toJson()).toList(),
        },
      ),
    );
    return response;
  }

  /// 修改电子围栏
  static Future<dynamic> updateFence(Fence fence) async {
    final response = await http.request(
      ReqOptions(
        path: '/hat/electronic/fence',
        method: 'PUT',
        data: {
          'fence': fence.toJson(),
          'coordinates': fence.coordinates?.map((e) => e.toJson()).toList(),
        },
      ),
    );
    return response;
  }

  /// 查询电子围栏详情
  static Future<Fence> getFenceById(String id) async {
    final response = await http.request(
      ReqOptions(path: '/hat/electronic/fence/$id', method: 'GET'),
    );
    return Fence.fromJson(response);
  }

  /// 删除电子围栏
  static Future<dynamic> deleteFence(String id) async {
    final response = await http.request(
      ReqOptions(path: '/hat/electronic/fence/$id', method: 'DELETE'),
    );
    return response;
  }
}
